src/EVEMon.Common/Models/BaseCharacter.cs

```diff
@@ -38,7 +38,18 @@ public abstract class BaseCharacter
         /// <exception cref="System.ArgumentNullException">skill</exception>
         public virtual float GetBaseSPPerHour(StaticSkill skill)
         {
-            return GetOmegaSPPerHour(skill);
+            float spPerHour = GetOmegaSPPerHour(skill);
+            Character chr = this as Character;
+            if (this is CharacterScratchpad && chr == null)
+            {
+                chr = (this as CharacterScratchpad).GetSourceCharacter();
+            }
+            if(chr != null)
+            {
+                spPerHour *= chr.CharacterStatus.TrainingRate;
```

Comment:
It may be more effective to make `AccountStatus` the overridden property instead, which would allow you to also collapse the `instanceof` chain here too.